services/compute/batchproc/controls.yaml

```
@@ -0,0 +1,50 @@
+common_controls:
+  - CCC.C01 # Prevent unencrypted requests
+  - CCC.C02 # Ensure data encryption at rest for all stored data
+  - CCC.C04 # Log all access and changes
+  - CCC.C05 # Prevent access from untrusted entities
+  - CCC.C06 # Prevent deployment in restricted regions
+  - CCC.C09 # Prevent tampering, deletion, or unauthorized access to access logs
+  - CCC.C10 # Prevent data replication to destinations outside of defined trust perimeter
+
+controls:
+  - id: CCC.BatchProc.C01 # Disable External IP Addresses for Compute Resources
+    title: Disable External IP Addresses for Compute Resources
+    objective: |
+      Ensure that compute resources used in batch processing tasks do not have external IP addresses to prevent unauthorized external access and enhance security.
+    control_family: Network Security
+    threats:
+      - CCC.TH01 # Access control is misconfigured
+      - CCC.TH02 # Data is intercepted in transit
+      - CCC.TH03 # Deployment region network is untrusted
+    nist_csf: PR.AC-5 # Network integrity is protected
+    control_mappings:
+      NIST_800_53:
+        - SC-7 # Boundary Protection
+        - SC-8 # Transmission Confidentiality and Integrity
+    test_requirements:
+      - id: CCC.BatchProc.C01.TR01
+        text: |
+          Attempt to access the compute resource over the public internet using an external IP address and verify that access is denied.
```

```suggestion
          When an external IP address attempts to access the compute resource over the public internet, the system MUST deny the request.
```

Adjusting to follow the recently proposed format of `When <activity>, <actor> MUST <action>`.

On another note... This seems like a common control for pretty much any networked red/amber resource?

---

yeah, I thought about that as well. I was thinking about how on the validation side of things, things could be different dependent on the service, so for now I left it in there, especially because there was not a direct common control for public external ip stuff right now